calculator/calculator.py

```diff
@@ -1,15 +1,151 @@
 import re
+import math
+from typing import Union
+from statistics import mean
 
 
-class Calculator:
+__TOKEN_SPECIFICATION = {
+    'ADDITION': r'(?P<ADDITION>\+)',
+    'SUBTRACTION': r'(?P<SUBTRACTION>-)',
+    'MULTIPLICATION': r'(?P<MULTIPLICATION>\*)',
+    'DIVISION': r'(?P<DIVISION>/)',
+    'POWER': r'(?P<POWER>\^)',
+    'NEGATE': r'(?P<NEGATE>~)',
+    'MODULE': r'(?P<MODULE>%)',
+    'FACTORIAL': r'(?P<FACTORIAL>!)',
+    'AVERAGE': r'(?P<AVERAGE>@)',
+    'MAX': r'(?P<MAX>\$)',
+    'MIN': r'(?P<MIN>&)',
+    'LEFT_PARENTHESIS': r'\(',
+    'RIGHT_PARENTHESIS': r'\)',
+    'NUMBER': '\d+(?:\.\d+)?',
+    'NUMBER_NEGATIVE_OPT': '-?\d+(?:\.\d+)?',
```

Comment:
Why it's not the only number? 
The regex here is: positive number or negative number => number!

Reply:
I have two cases:
1. I search for a number that can be positive or negative.
2. I search for number that must be positive like in: -5! which analyzed as X minus 5! INSTEAD of taking the minus into the factorial which is not defined for negative numbers.